pkg/sql/plan/function/builtin/multi/substr_test.go

```diff
@@ -434,6 +434,211 @@ func TestSubStrUTF(t *testing.T) {
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test11",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -4, 0, false, procs),
+			proc:       procs,
+			wantBytes:  []byte("klmn"),
+			wantScalar: true,
+		},
+		{
+			name:       "Test12",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -14, 0, false, procs),
+			proc:       procs,
+			wantBytes:  []byte("abcdefghijklmn"),
+			wantScalar: true,
+		},
+		{
+			name:       "Test13",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -16, 0, false, procs),
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test14",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -4, 3, true, procs),
+			proc:       procs,
+			wantBytes:  []byte("klm"),
+			wantScalar: true,
+		},
+		{
+			name:       "Test15",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -14, 10, true, procs),
+			proc:       procs,
+			wantBytes:  []byte("abcdefghij"),
+			wantScalar: true,
+		},
+		{
+			name:       "Test16",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -14, 15, true, procs),
+			proc:       procs,
+			wantBytes:  []byte("abcdefghijklmn"),
+			wantScalar: true,
+		},
+		{
+			name:       "Test17",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -16, 10, true, procs),
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test18",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -16, 20, true, procs),
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test19",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -16, 2, true, procs),
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test20",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -12, 2, true, procs),
+			proc:       procs,
+			wantBytes:  []byte("cd"),
+			wantScalar: true,
+		},
+		{
+			name:       "Test21",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -12, 14, true, procs),
+			proc:       procs,
+			wantBytes:  []byte("cdefghijklmn"),
+			wantScalar: true,
+		},
+		{
+			name:       "Test22",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -12, 0, true, procs),
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test23",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -6, -5, true, procs),
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test24",
+			vecs:       makeSubStrBlobVectors([]byte("abcdefghijklmn"), -6, -10, true, procs),
+			proc:       procs,
+			wantBytes:  []byte(""),
+			wantScalar: true,
+		},
+		{
+			name:       "Test25",
```

Comment:
It is always better to create a longer string for testing.   Something like 50 chars long.